Core.Collectors/Config/ConfigManager.cs

```diff
@@ -156,6 +161,32 @@ protected JToken GetAuthenticationToken(string collectorType)
             return authenticationToken;
         }
 
+        public JToken GetApiDomainToken()
+        {
+            ValidateSettingsExist();
+
+            if (!this.apiDomainTokenMap.TryGetValue(DefaultKey, out JToken apiDomainToken))
+            {
+                apiDomainToken = this.apiDomainTokenMap[DefaultKey];
+            }
+            return apiDomainToken;
+        }
+
+        public string GetApiDomain()
+        {
+            string apiDomain = string.Empty;
+            JToken apiDomainToken = this.GetApiDomainToken();
+            try
+            {
+                apiDomain = apiDomainToken.ToString();
+            }
+            catch (Exception)
+            {
+                throw new FatalTerminalException($"Invalid URI: The hostname could not be parsed for API domain {apiDomainToken}");
```

Comment:
Can we change the error message to be something more descriptive here?

Reply:
Yes, I added extra details. "The API domain must be provided in Settings.json under the github-settings Azure Blob.".

Reply:
Nit: since this is functionality in _CEDAR.Core.Collector_, we cannot assume that this will be used only in _CEDAR.GitHub.Collector_. Let's say we have collectors for ADO or GitLab.

Reply:
Right, I didn't think about that. Would you say that there is sufficient information for a dev to just add "The API domain must be provided in Settings.json." If I had that error message (for local settings) when I was trying to get the collectors running locally I would have had enough info to debug it immediately.

Reply:
Would be totally fine for me, too. You can go overboard and even link to the documentation if you want to, but I agree that this would already be sufficient enough.

Reply:
Great, thanks! I just updated the PR.